### DISCUSSION TITLE
Can you explain what Sync does and how to use it?

### DISCUSSION THREAD

**Discussion by gaveitatry2** [violentmonkey/violentmonkey#1155] [Q&A]
I asked on Reddit if there was a way to upload my scripts online but make them private so that no one else can see or use them.  The idea was to have all my laptops access that online version so all userscripts will be updated at the same time.  That way I wouldn't have to copy and paste changes from one to the other.  And this was less important, but maybe this way I won't risk losing my userscripts in case I forget and uninstall Violentmonkey.  One Reddit user told me that Tampermonkey had a Sync feature but he didn't go into detail on what it does exactly.  So anyway, I glanced at ViolentMonkey's Settings page and it seems to have Sync.  And it had Google Drive and Dropbox.  I'm not sure if I still have a Dropbox account but I definitely do have a Google Drive account.  How does this work?  If I want to make changes to a userscript and have all laptops get the same update, does that mean that instead of editing the userscript in the browser I need to edit it elsewhere and then upload the new version to Google Drive?  Or can I edit from Google Drive directly?  Or can I edit from the ViolentMonkey extension but it still update everywhere?  I'm just so confused.  I don't know anything about this.  That's why I need someone to explain it to me like I am five years old and cover ALL the details so I can see if this is something that will be a benefit to me and worth the hassle.  Or maybe I should just stick with copying and pasting and making periodic .zip export backups.

**Comment by tophf** [comment]
It works just like it does in any other app for the last decade. You enable it and the scripts are synchronized between the computers where you have Violentmonkey with sync option enabled for the same service. When you edit a script on such a computer, the updated code is sent to the online service. Next time you connect on another computer, it will receive the updated code.

Under the hood, when you enable `sync` Violentmonkey creates a special file in the online service. You don't need to edit it. In some services it's not even visible as a file, but you can see it in a special place in settings, for example.

**Comment by gera2ld** [comment]
FYI you cannot edit from Google Drive directly because Violentmonkey only has access to [application data folder](https://developers.google.com/drive/api/v3/appdata) of your Google Drive as recommended [here](https://developers.google.com/drive/api/v3/about-auth). tldr the application data is hidden from the user and other applications.

Nevertheless, you can see all the files created by Violentmonkey in Dropbox.
